src/client/components/fields/DateField.tsx

```diff
@@ -8,46 +8,64 @@ import {
   FormHelperText,
   FormErrorMessage,
   Input,
+  InputGroup,
+  InputRightElement,
 } from '@chakra-ui/react'
 
 import 'react-datepicker/dist/react-datepicker.css'
 import '../../styles/date-picker.css'
 
 import { Field } from '../../../types/checker'
+import { BiCalendar } from 'react-icons/bi'
+
+interface DatePickerInputProps {
+  value: string
+  onClick: (event: React.MouseEvent<HTMLInputElement, MouseEvent>) => void
+}
+
+const DatePickerInput: FC<DatePickerInputProps> = ({ value, onClick }) => (
+  <InputGroup>
+    <Input
+      type="text"
+      placeholder="DD/MM/YYYY"
+      className="react-datepicker-ignore-onclickoutside"
+      value={value}
+      onClick={onClick}
+    />
+    <InputRightElement pointerEvents="none" children={<BiCalendar />} />
+  </InputGroup>
+)
 
 export const DateField: FC<Field> = ({ id, title, description }) => {
   const styles = useStyles()
-  const { register, errors } = useFormContext()
-  const [date, setDate] = useState(new Date())
+  const { control, errors } = useFormContext()
   const error = errors[id]
 
-  const changeHandler = (newDate: Date) => {
-    if (newDate) setDate(newDate)
-  }
-
   return (
-    <FormControl isInvalid={error} id={id}>
-      <FormLabel sx={styles.label} htmlFor={id}>
-        {title}
-      </FormLabel>
-      {description && <FormHelperText mb={4}>{description}</FormHelperText>}
-      <Input
-        type="hidden"
-        name={id}
-        value={date ? date.valueOf() : undefined}
-        ref={register({ required: true })}
-      />
-      <DatePicker
-        wrapperClassName={error ? 'fieldError' : ''}
-        value={date ? date.toISOString().split('T')[0] : ''}
-        selected={date}
-        onChange={changeHandler}
-        showPopperArrow={true}
-        showMonthDropdown
-        showYearDropdown
-        dropdownMode="select"
-      />
-      <FormErrorMessage>Field is required</FormErrorMessage>
-    </FormControl>
+    <Controller
+      name={id}
+      control={control}
+      render={({ onChange, value }, { invalid }) => (
+        <FormControl isInvalid={invalid} id={id}>
+          <FormLabel sx={styles.label} htmlFor={id}>
+            {title}
+          </FormLabel>
+          {description && <FormHelperText mb={4}>{description}</FormHelperText>}
+          <DatePicker
```

Comment:
Was wondering if we are able to implement the multiple level of selection in the design from Nat. From what I gather from the `react-datepicker` docs, perhaps we can use `renderCustomHeader` + `showMonthYearPicker` + `showYearPicker` to achieve it.

<img src="https://user-images.githubusercontent.com/3666479/109470566-90f5f800-7aaa-11eb-9662-261c41c7bb99.png" width=400 />

<img src="https://user-images.githubusercontent.com/3666479/109470598-9c492380-7aaa-11eb-9152-e86a8dae7600.png" width=400 />